### PR TITLE
Feat#3_integrate-husky-commitlint

### PR DESCRIPTION
**Pull request by CasuallyCaffeinated**
#### Namely, I passed in the following test messages into my terminal to ensure that everything worked as expected:
* #### `echo 'testing commitlint, this should fail' | npx commitlint`

>      ✖   found 2 problems, 0 warnings
>      ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

#### After that, I passed in a test message that would not fail:
* #### `echo 'feat: test message' | npx commitlint`
#### I elected to use the popular `@commitlint/config-conventional` setup with our configuration file. If you'd like to learn more about what that means, feel free to check out eithers of these docs: 
* [Commitlint Docs](https://commitlint.js.org/#/) || [Conventional Commits Docs](https://www.conventionalcommits.org/en/v1.0.0/). 

#### Knowing, however that documentation can be boring, what's most important to keep in mind is to add a `type` to the beginning of your commits. A `type` is string that's part of the `type-enum`. You can find out more about the `type-enum` [here.](https://github.com/conventional-changelog/commitlint/tree/master/@commitlint/config-conventional#type-enum) If you have any questions, feel free to reach out.
* #### Example: `"feat: created a textbox component"`

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx --no-install commitlint --edit
```
Comment (aetheode) [inline .husky/commit-msg:4]: I'd actually  never seen "no-install" before

https://docs.npmjs.com/cli/v7/commands/npx#compatibility-with-older-npx-versions 
> The --no-install option is deprecated, and will be converted to --no.

Comment (CasuallyCaffeinated) [reply .husky/commit-msg:4]: I remember coming across something about it being deprecated. But there were a few conflicting sources I found so I just left it as is. But yeah, I can go ahead and change it.

Comment (CasuallyCaffeinated) [reply .husky/commit-msg]: I pushed up the new change. I did however encounter a weird error that I had to bypass once I changed the Husky commit-msg script from `--no-install` to `--no`.   

The error I was getting was this: 
>→ No staged files match any configured task.
> @commitlint/cli@16.2.1 - Lint your commit messages

> [input] reads from stdin if --edit, --env, --from and --to are omitted

> [input] is required: supply via stdin, or --env or --edit or --from and --to
> husky - commit-msg hook exited with code 1 (error)

The only way I could bypass it was to pass in the `--no-verify` flag to my commit message. Still, that defeats the purpose of commitlint and Husky if we're not going to verify our commits. Let me know if you also get a similar error with commitlint whenever this branch gets pushed into main. If so, the new `--no` flag might be the issue.

**Comment by OleDakotaJoe** [comment]
Try adding $1 to the commit-msg hook

Should be `npx --no commitlint --edit $1`

**Comment by CasuallyCaffeinated** [comment]
> Try adding $1 to the commit-msg hook 
> Should be `npx --no commitlint --edit $1`

Got it, will do. Thanks! It must've slipped my mind, sorry about that.

File: .husky/commit-msg (hunk 1 above)
Comment (aetheode) [reply .husky/commit-msg]: @CasuallyCaffeinated  - Yeah that is interesting. I'd actually be worried to merge this PR if that problem exists as we would need to communicate with the entire team to also use the --no-verify flag to unblock them

If you don't get an error with --no-install, let's roll with that for now? and I'll investigate it after it merges

**Comment by CasuallyCaffeinated** [comment]
> Try adding $1 to the commit-msg hook
> 
> Should be `npx --no commitlint --edit $1`

File: .husky/commit-msg (hunk 1 above)
Comment (CasuallyCaffeinated) [reply .husky/commit-msg]: @aetheode Yeah, I tried one more experimental change. Whenever the flag `--no` was present, I was unable to commit. However, when I switched it to `--no-install` it functioned properly. I even tried what @OleDakotaJoe had suggested, namely adding `$1` to the end of the command, but that didn't solve it either. So for now, I'd say we need to stick to the flag `--no-install`.

**Comment by OleDakotaJoe** [comment]
Definitely just go-ahead and use the --no-install

As soon as you get this pushed up we will merge

**Comment by CasuallyCaffeinated** [comment]
Pushed up the fix. Ready to be merged whenever we decide.

**Comment by OleDakotaJoe** [comment]
merged

**Pull request merged**